frontend/src/pages/SearchPage.tsx

```diff
@@ -0,0 +1,60 @@
+import { useSearchRestaurants } from '@/api/RestaurantApi';
+import SearchBar, { SearchForm } from '@/components/SearchBar';
+import SearchResultCard from '@/components/SearchResultCard';
+import SearchResultInfo from '@/components/SearchResultInfo';
+import { useState } from 'react';
+import { useParams } from 'react-router-dom';
+
+export type SearchState = {
+  searchQuery: string;
+};
+
+const SearchPage = () => {
+  const { city } = useParams<{ city: string }>();
+  const [searchState, setSearchState] = useState<SearchState>({
+    searchQuery: '',
+  });
+  const { results, isLoading } = useSearchRestaurants(searchState, city);
```

Comment:
<div>

<strong>Bito Code Review Agent Run</strong> <a href=https://github.com/omkardongre/Food-Ordering-App/pull/43#issuecomment-2408154714>#86982f</a><span> - 10/11/2024, 09:46 pm</span>
<br>

<div><small>🔴</small> High importance</div>

<div id="suggestion">
<div id="issue">
<b>Issue:</b> The useSearchRestaurants hook is called on every render, which may lead to unnecessary API calls and potential performance issues.
</div>
<div id="fix">
<b>Fix:</b> Implement useMemo hook to memoize the useSearchRestaurants call, preventing unnecessary API calls on re-renders.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div id="code">

```
 @@ -14,7 +14,10 @@
    const [searchState, setSearchState] = useState<SearchState>({
      searchQuery: '',
    });
 -  const { results, isLoading } = useSearchRestaurants(searchState, city);
 +  const { results, isLoading } = useMemo(() => 
 +    useSearchRestaurants(searchState, city),
 +    [searchState, city]
 +  );
 
    const setSearchQuery = (searchFormDate: SearchForm) => {
      setSearchState((prevState) => ({
```
</div>
</details>
</div>
<br>


</div>

Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged